### PR TITLE
[1232] Providers add edit accrediting provider step 1 js version

### PR DESCRIPTION
**Pull request by defong**
### Context
Providers add edit accrediting provider step 1 js version 

### Changes proposed in this pull request

Refactored so the js is not crazy and untested

### Guidance to review

image

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally
- [ ] Inform data insights team due to database changes

### CODE REVIEW

**Comment by defong** [comment]
[Providers: Add & Edit Accrediting Provider - Step 1 (JS version)](https://trello.com/c/PtEGPcpL/1232-providers-add-edit-accrediting-provider-step-1-js-version)

**Review by goodviber — APPROVED** [review]
Looks nice, just one comment.

File: app/javascript/publish/autocomplete/autocomplete.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        templates: {
+          inputValue,
+          suggestion
+        },
+        onConfirm: onConfirm(input),
+        confirmOnBlur: false,
+        autoselect: true
+      })
+
+      // Hijack the original input to submit the selected provider_code.
```
Comment (goodviber) [inline app/javascript/publish/autocomplete/autocomplete.js:31]: Is this still required?

Comment (defong) [reply app/javascript/publish/autocomplete/autocomplete.js]: yes

**Review by victoriawinlow — APPROVED** [review]

**Review by darokel — APPROVED** [review]

**Pull request merged**